tools/verify-fs.in

```diff
@@ -94,9 +94,30 @@ for ROOTFS in *.squashfs; do
         log "Missing MD5 checksum file: '${ROOTFS}.md5'"
     fi
 
-    if [ -s "${ROOTFS}.sha256" ] ; then
+    if [ -s "${ROOTFS}.sha256" ] ; then 
         if [ /usr/bin/sha256sum -c "${ROOTFS}.sha256" 2>/dev/null ] ; then
-            log "Good integrity of ${ROOTFS}, SHA256 checksum is valid"
+            if [ -s "${ROOTFS}-manifest.json" ] ; then
+                    SHA256_COMPUTED=$(cut -d ' ' -f1 < "${ROOTFS}.sha256")
+                    SHA256_MANIFEST=$(grep ShaChecksum "${ROOTFS}-manifest.json" | cut -d '"' -f4)
+
+                if [ "${SHA256_COMPUTED}" = "${SHA256_MANIFEST}" ]; then
+                    if [ -s "${ROOTFS}-manifest.json.p7s" ] ; then
+                        SIGNED_BY=$(grep SignedBy "${ROOTFS}-manifest.json" | cut -d '"' -f4)
+
+                        if [ openssl cms -verify -binary -inform der -CAfile /usr/share/authorities/${SIGNED_BY}/rootCa1.pem -purpose any -out /dev/null -content "${ROOTFS}-manifest.json" -in "${ROOTFS}-manifest.json.p7s" ] ; then
+                            log "Image signature is valide for ${ROOTFS}"
+                        else
+                            log "Image signature is invalide for ${ROOTFS}"
+                        fi
+                    else
+                        log "Manifest signature not found for ${ROOTFS}"
+                    fi
+                else
+                    log "Manifest sha256 do not match with the sha256 for ${ROOTFS}"
```

Comment:
Better message would be `Manifest sha256 hash does not match with the sha256 hash for ${ROOTFS}`